packages/plugins/src/hardhat/tasks.ts

```diff
@@ -1375,3 +1378,140 @@ export const listProjectsTask = async ({}, hre: HardhatRuntimeEnvironment) => {
+
+  const ChugSplashManager = getChugSplashManager(
+    signer,
+    parsedConfig.options.projectName
+  )
+
+  const proposers = []
+
+  // Fetch current owner
+  const owner = await getProjectOwnerAddress(
+    provider,
+    parsedConfig.options.projectName
+  )
+  proposers.push(owner)
+
+  // Fetch all previous proposers
+  const addProposerEvents = await ChugSplashManager.queryFilter(
+    ChugSplashManager.filters.ProposerAdded()
+  )
+
+  // Verify if each previous proposer is still a proposer before adding it to the list
+  for (const proposerEvent of addProposerEvents) {
+    const address = proposerEvent.args.proposer
+    const isStillProposer = await ChugSplashManager.proposers(address)
+    if (isStillProposer && !proposers.includes(address)) {
+      proposers.push(address)
+    }
+  }
+
+  // Display the list of proposers
+  displayProposerTable(proposers)
+}
+
+task(TASK_CHUGSPLASH_LIST_PROPOSERS)
+  .setDescription('Lists all of the approved proposers for this project')
+  .addPositionalParam(
+    'configPath',
+    'Path to the ChugSplash config file to propose'
+  )
+  .setAction(listProposersTask)
+
+export const addProposerTask = async (
+  args: {
+    configPath: string
+    newProposers: string[]
+  },
+  hre: HardhatRuntimeEnvironment
+) => {
+  const { configPath, newProposers } = args
+
+  if (newProposers.length === 0) {
+    throw new Error('You must specify at least one proposer to add.')
+  }
+
+  const parsedConfig = loadParsedChugSplashConfig(configPath)
+  const provider = hre.ethers.provider
+  const signer = provider.getSigner()
+
+  const spinner = ora()
+
+  if (
+    (await isProjectRegistered(signer, parsedConfig.options.projectName)) ===
+    false
+  ) {
+    errorProjectNotRegistered(
+      await getChainId(hre.ethers.provider),
+      hre.network.name,
+      configPath
+    )
+  }
+
+  const ChugSplashManager = getChugSplashManager(
+    signer,
+    parsedConfig.options.projectName
+  )
+
+  // Fetch current owner
+  const projectOwnerAddress = await getProjectOwnerAddress(
+    provider,
+    parsedConfig.options.projectName
+  )
+  if (projectOwnerAddress !== (await signer.getAddress())) {
+    throw new Error(`Project is owned by: ${projectOwnerAddress}.
+You attempted to add a proposer using address: ${await signer.getAddress()}`)
+  }
+
+  for (const newProposer of newProposers) {
+    spinner.start(`Adding proposer ${newProposer}...`)
+
+    const isAlreadyProposer = await ChugSplashManager.proposers(newProposer)
+    if (isAlreadyProposer) {
+      throw new Error(
+        `A proposer with the address ${newProposer} has already been added.`
+      )
+    }
+
+    await (await ChugSplashManager.addProposer(newProposer)).wait()
+
+    spinner.succeed(`Proposer ${newProposer} successfully added!`)
+  }
+
+  await listProposersTask({ configPath }, hre)
+}
+
+task(TASK_CHUGSPLASH_ADD_PROPOSER)
+  .setDescription('Adds a new proposer to the list of approved proposers')
+  .addPositionalParam(
```

Comment:
So I agree with you that it would be good for there to be only one positional parameter per task. However, I don't think it's a good idea for us to include `configPath` and not make it the positional param simply because it is a positional param in all of our other tasks. So it's very unintuitive that it would not be positional in this specific task. 

Unfortunately, I don't think we can make a non-positional and variadic param. So I believe our choices are to either keep it as is or only allow the user to input a single proposer address per task run.